File: blockSearch.js
//check for min element
function checkMin(a,b)
  {
    let c=a>b?a:b;
    return c;
  }
function search(ar,item)
{
  prev=0;
  let n=ar.length;
  let step=Math.floor(Math.sqrt(n));
  console.log(step);
  while(ar[checkMin(step,n)-1]<item)
  {
    prev=step;
    step+=Math.floor(Math.sqrt(n));
    if(prev>=n) //element not found
      {
        return;
      }
  }
  
  while(a[prev] < item)
    {
      prev++;
      if(a[prev]==item)
        {
          return prev;
        }
      if(prev == checkMin(step,n))
        {
          return;
        }
    }
}


// test cases
var inputArray = [];
var size; //Maximum Array size
size=prompt("enter size ");
for(var i=0; i<size; i++) {
	
	//Taking Input from user
	inputArray[i] = prompt('Enter Element ' + (i+1));
}
console.log(inputArray);
let item=prompt('Enter item ');
//Print the array in the console.
let result=search(inputArray,item);
if(!result)
  {
    console.log("out of range");
  }
else
  {
    
console.log(result);
    
  }

File: blocksearch.go
package main
import (
	"fmt"
	"math"
)
func jumpSearch(arr []int,x int,n int) int{
	var step=int(math.Sqrt(float64(n)))
	var prev=0
	for arr[int(math.Min(float64(step),float64(n))-1)]<x{
		prev=step
		step+=int(math.Sqrt(float64(n)))
		if prev>=n{
			return -1
		}
	}
	for arr[prev]<x{
		prev++
		if prev==int(math.Min(float64(step),float64(n))){
			return -1
		}
	}
	if arr[prev]==x{
		return prev
	}
	return -1
}
func main(){
	var n,key int
	var array []int
	fmt.Printf("Enter the size of the array: ")
	fmt.Scanf("%d",&n)
	array=make([]int,n)
	fmt.Printf("Enter the sorted element in the array:\n")
	for i:=0;i<n;i++{
		fmt.Scanf("%d ",&array[i])
	}
	fmt.Printf("\nEnter the element to be searched: ")
	fmt.Scanf("%d",&key)
	fmt.Println("Fount at index",jumpSearch(array,key,n))
}
